### DISCUSSION TITLE
When creating containers in loop, can we name them separately so that we can use them later?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#1297] [Q&A] [answered]
### Question

Hi All,

I have a list, based on which I am creating rows. My challenge is later at some point, I need to access specific rows, and do stuff inside there, like add some content, or remove, etc.

Currently I am using the ```enumerate``` method to just index through the list options and then create a ```if-then-else``` condition. Like this below:
```python
list = ['A', 'B', 'C', 'D']

with ui.column() as parent:
    for item in list:
        with ui.card():
            ui.label(item)

for index, item in enumerate(parent):
    if index == 2:
        with item:
            ui.label('I can add more stuff here')
```

Is this the right way to do this? Or is there a way to directly assign names or ids to the ```ui.card()``` elements like maybe card1, card2 etc. so that later I can simply do ```with card1``` and no need to enumerate through the whole thing?

Please let me know what would be the best approach to this problem.

Thanks a lot,
Anindya

**Comment by falkoschindler** [comment, marked as answer]
Since you create the cards based on a string, you might want to put them into a dictionary which you can access like `cards['C']`:
```py
list = ['A', 'B', 'C', 'D']

cards = {}
with ui.column():
    for item in list:
        with ui.card() as cards[item]:
            ui.label(item)

with cards['C']:
    ui.label('I can add more stuff here')
```

Or you use the fact that each container element holds a list of children:
```py
list = ['A', 'B', 'C', 'D']

with ui.column() as parent:
    for item in list:
        with ui.card():
            ui.label(item)

with parent.default_slot.children[2]:
    ui.label('I can add more stuff here')
```

**Reply by Anindya088** [reply]
Hi @falkoschindler ,

Thanks a ton ! This is exactly what I was looking for, so great!

I should have tried the dict idea, did not occur to me that the ui elements can be put inside a dict like that, great to know that.
But I love the 2nd method where simply I can point to the nth child and do my thing. That is very helpful. 

I got one question, this ```default_slot``` is this a Quasar thing? Because I did not see this in the ```ui.card``` docs, so is it a generic property of the Quasar q-card that you are applying here? 

Thanks so much for all the quick helps and solutions.
Anindya

**Reply by falkoschindler** [reply]
Yes, the `default_slot` is not well documented, since you usually don't need to know about it. The thing is, Vue components can have children in the so-called default slot. But they can also have other sets of children in other "slots". Therefore NiceGUI elements organize their children in slots as well. If you just say `with ui.row(): ui.label()`, you're adding a label to the row's default slot. And if you say `with table.add_slot('top-right'): ...`, you're adding elements to the "top-right" slot of the table.

**Reply by Anindya088** [reply]
Understood, thanks so much for explaining this. I'll spend more time on Quasar I think. There's a lot to learn over there which gets transferred here on NiceGUI and Python 😄
